src/app/rpmostree-pkg-builtins.c

```diff
@@ -157,7 +158,37 @@ rpmostree_builtin_pkg_add (int            argc,
     }
 
   for (int i = 1; i < argc; i++)
-    g_ptr_array_add (packages_to_add, argv[i]);
+    {
+      const char *pkgspec = argv[i];
+      g_autofree char *abspath = NULL;
+
+      if (!g_str_has_suffix (pkgspec, ".rpm") || /* repo install */
+           g_str_has_prefix (pkgspec, "/"))      /* local install */
+        {
+          g_ptr_array_add (packages_to_add, g_strdup (pkgspec));
+          continue;
+        }
+
+      /* OK, it's a relative path, we need to check that it
+       * exists and canonicalize it for the daemon. */
```

Comment:
Nice, I didn't realize D-Bus supported this. I'd like to try it out, though it's a bit unfortunate that it'd have to be hidden away in the `options` array if we want to avoid adding another method.